server/middleware/populateCurrentUserRoles.ts

```diff
@@ -0,0 +1,12 @@
+import { RequestHandler } from 'express'
+import jwtDecode from 'jwt-decode'
+
+export default function populateCurrentUserRoles(): RequestHandler {
+  return async (req, res, next) => {
+    if (res.locals.user?.token) {
+      const { authorities: roles } = jwtDecode(res.locals.user.token) as { authorities?: string[] }
+      res.locals.user = { roles, ...res.locals.user }
```

Comment:
I guess you could just include this line in `server/middleware/authorisationMiddleware.ts` where the token is already decoded and then you wouldn't need to have this file at all?

But, appreciate that it it makes it clearer what is going on, alongside the `populateCurrentUser` middleware...

Reply:
will leave it in for now.